chain/chunks/src/lib.rs

```diff
@@ -623,21 +631,46 @@ impl ShardsManager {
             let old_block = header_head.last_block_hash != parent_hash
                 && header_head.prev_block_hash != parent_hash;
 
-            let request_result = self.request_partial_encoded_chunk(
-                height,
-                &parent_hash,
-                shard_id,
-                &chunk_hash,
-                false,
-                old_block,
-                fetch_from_archival,
-            );
-            if let Err(err) = request_result {
-                error!(target: "chunks", "Error during requesting partial encoded chunk: {}", err);
+            // If the protocol version is later than the one where chunk forwarding was introduced,
+            // we purposely do not send chunk request messages right away for new blocks. Such requests
+            // will eventually be sent because of the `resend_chunk_requests` loop. However,
+            // we want to give some time for any `PartialEncodedChunkForward` messages to arrive
+            // before we send requests.
+            #[cfg(feature = "forward_chunk_parts")]
+            let protocol_version_check = _protocol_version
+                < PROTOCOL_FEATURES_TO_VERSION_MAPPING[&ProtocolFeature::ForwardChunkParts];
+            #[cfg(not(feature = "forward_chunk_parts"))]
+            let protocol_version_check = true;
```

Comment:
The strict less is correct here. If the protocol version is greater than or equal to the forward chunks protocol version then we want to skip the block of code because it is not needed.

I agree it is somewhat awkward and error-prone though, as written. I thought we were going to pivot away from explicit protocol version checks in the main code and instead have a variable like `is_chunk_forwarding_enabled` which is set using a function (more macro) that depends on the protocol version. Then the `if` statement here would be 

```rust
if !is_chunk_forwarding_enabled || fetch_from_archival || old_block {
    ...
}
```

which I think is more clear to someone reading the code.

Reply:
@birchmd I see your point about the `<` vs `<=`. I clearly did not understand the logic, which is why I think it is good to have the macro here, so it would look like:

```rust
let is_chunk_forwarding_enabled = check_feature!(ForwardChunkParts, _protocol_version).unwrap_or(false);
```

I wonder if we need an `Option<bool>` there, though. I think we can start with a simple `bool`. :thinking:

```rust
let is_chunk_forwarding_enabled = check_feature!(ForwardChunkParts, _protocol_version);
```